src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class FigureSupplier {
+    public Figure getRandomFigure(int maxSize) {
+        Random rand = new Random();
+        String color = ColorSupplier.getRandomColor();
+        int randFigureIndex = rand.nextInt(AvailableFigure.values().length);
+        switch (AvailableFigure.values()[randFigureIndex].name()) {
+            case "RECTANGLE":
+                int randRightSide = rand.nextInt(maxSize);
+                int randTopSide = rand.nextInt(maxSize);
```

Comment:
Instead of passing expressions directly in the constructor, create separate variables for 'randRightSide' and 'randTopSide' and then pass them to improve code readability.